### PR TITLE
[Communication] - Chat - Replaced ChatUserCredentialPolicy with BearerTokenAuthenticationPolicy

### PR DESCRIPTION
**Pull request by jbeauregardb**
This PR removes a custom policy built for the Chat SDK and replaces it with a generic BearerTokenAuthenticationPolicy to keep consistency with other SDKs.

TODO:

- [x] CommunicationTokenCredential tests

### CODE REVIEW

**Comment by JoshuaLai** [comment]
Everything looks good but I would recommend removing `ChatUserCredentialPolicy` I don't think there are any more instances of it being used

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java [hunk 1]
```diff
@@ -166,9 +168,10 @@ public ChatAsyncClient buildAsyncClient() {
         } else {
             Objects.requireNonNull(communicationUserCredential);
             Objects.requireNonNull(httpClient);
+            CommunicationTokenCredential tokenCredential = new CommunicationTokenCredential(communicationUserCredential);
 
             pipeline = createHttpPipeline(httpClient,
-                new ChatUserCredentialPolicy(communicationUserCredential),
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java:171]: Looks like you should also remove the ChatUserCredentialPolicy file as it is not used anymore

Comment (jbeauregardb) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java:171]: Just removed it!

Comment (ankitarorabit) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java:171]: I see you already took care of it

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.concurrent.ExecutionException;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * This class serves as aCommunicationUserCredential wrapper that
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java:15]: nit: space between "a CommunicationUserCredential"

**Review by ankitarorabit — APPROVED** [review]

**Review by JianpingChen — APPROVED** [review]
Ideally, there should be at least one test case for the builder

**Pull request merged**